signature/cose/envelope_test.go

```diff
@@ -79,144 +39,177 @@ func TestParseEnvelopeError(t *testing.T) {
 	}
 }
 
-// Tests various scenarios around generating a COSE envelope
 func TestSign(t *testing.T) {
-	env := NewEnvelope()
+	env := envelope{}
 	for _, signingScheme := range signingSchemeString {
-		t.Run(fmt.Sprintf("with %s scheme when all arguments are present", signingScheme), func(t *testing.T) {
-			signRequest, err := newSignRequest(signingScheme)
-			if err != nil {
-				t.Fatalf("newSignRequest(). Error = %s", err)
-			}
-			_, err = env.Sign(signRequest)
-			if err != nil {
-				t.Fatalf("sign failed. Error = %s", err)
+		for _, keyType := range keyType {
+			var size []int
+			if keyType == signature.KeyTypeRSA {
+				size = []int{2048, 3072, 4096}
+			} else {
+				size = []int{256, 384, 521}
 			}
-		})
+			for _, size := range size {
+				t.Run(fmt.Sprintf("with %s scheme, %v keyType, %v keySize when all arguments are present", signingScheme, keyType, size), func(t *testing.T) {
+					signRequest, err := newSignRequest(signingScheme, keyType, size)
+					if err != nil {
+						t.Fatalf("newSignRequest() failed. Error = %s", err)
+					}
+					_, err = env.Sign(signRequest)
```

Comment:
should we do some checking on the returned signature envelope bytes?

Reply:
code updated